test/integration/filters/repick_cluster_filter.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace RepickClusterFilter {
+
+// The cluster names and cluster headers here need to be correlated correctly with
+// weighted_cluster_integration_test or any other end users, to make sure that header modifications
+// have been done on the correct target. So they are declared and defined here.
+inline constexpr absl::string_view ClusterName = "cluster_%d";
+inline constexpr absl::string_view ClusterHeaderName = "cluster_header_%d";
```

Comment:
This feels like it might be unnecessarily complicate these tests, how bad is it to just hard code the header modifications that it does?

Reply:
Doing this can enable generic testing approach (i.e. random combination of cluster_header + cluster_name).
I lay out some rational behind the generic approach. We can discuss further there. So I just resolve/close the conversation here.